cumulusci/tasks/bulkdata/dates.py

```diff
@@ -0,0 +1,115 @@
+    fields = mapping.get_field_list()
+
+    date_fields = [
+        fields.index(f)
+        for f in mapping.get_fields_by_type("date", org_config)
+        if f in mapping.fields
+    ]
+    date_time_fields = [
+        fields.index(f)
+        for f in mapping.get_fields_by_type("datetime", org_config)
+        if f in mapping.fields
+    ]
+
+    return (date_fields, date_time_fields)
+
+
+def adjust_relative_dates(
+    mapping,
+    context,
+    record: List[Optional[str]],
+    operation: DataOperationType,
+):
+    """Convert specified date and time fields (in ISO format) relative to the present moment.
+    If some date is 2020-07-30, anchor_date is 2020-07-23, and today's date is 2020-09-01,
+    that date will become 2020-09-07 - the same position in the timeline relative to today."""
+
+    date_fields, date_time_fields = context
+
+    # Determine the direction in which we are converting.
+    # For extracts, we convert the date from today-anchored to mapping.anchor_date-anchored.
+    # For loads, we do the reverse.
+
+    if operation is DataOperationType.QUERY:
+        current_anchor = date.today()
+        target_anchor = mapping.anchor_date
+    else:
+        current_anchor = mapping.anchor_date
+        target_anchor = date.today()
+
+    r = record.copy()
+
+    for index in date_fields:
+        if operation is DataOperationType.QUERY:
+            index += 1  # For the Id field.
+        if r[index]:
+            r[index] = date_to_iso(
+                _convert_date(target_anchor, current_anchor, iso_to_date(r[index]))
+            )
+
+    for index in date_time_fields:
+        if operation is DataOperationType.QUERY:
+            index += 1  # For the Id field.
+        if r[index]:
+            r[index] = salesforce_from_datetime(
+                _convert_datetime(
+                    target_anchor,
+                    current_anchor,
+                    datetime_from_salesforce(r[index]),
+                )
+            )
+
+    return r
+
+
+# The Salesforce API returns datetimes with millisecond resolution, but milliseconds
+# are always zero (that is, .000). Python does parse this with strptime.
+# Python renders datetimes into ISO8601 with microsecond resolution (.123456),
+# which Salesforce won't accept - we need exactly three digits, although they are
+# currently ignored. Python also right-truncates to `.0`, which Salesforce won't take.
+# Hence this clumsy workaround.
+
+
+def datetime_from_salesforce(d):
+    """Create a Python datetime from a Salesforce-style ISO8601 string"""
+    return datetime.strptime(d, "%Y-%m-%dT%H:%M:%S.%f%z")
+
+
+def salesforce_from_datetime(d):
+    """Create a Salesforce-style ISO8601 string from a Python datetime"""
+    return d.strftime("%Y-%m-%dT%H:%M:%S.{}+0000").format(
+        str(d.microsecond)[:3].ljust(3, "0")
+    )
+
+
+# Python 3.6 doesn't support the isoformat()/fromisoformat() methods. Shims.
```

Comment:
I think it supports isoformat but not fromisoformat

Reply:
Ah, you're right. I'll correct the comment. Do you have any concerns about keeping the shim? I ended up liking the explicitness given the other minor mismatches between `datetime` and the Salesforce API's behavior.

Reply:
![I'll allow it](https://media.giphy.com/media/SmoCFhZCi1kzu/giphy.gif)